service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/C21OrderEmailContentProviderTest.java

```diff
@@ -0,0 +1,148 @@
+                "167888", "/case/" + JURISDICTION + "/" + CASE_TYPE + "/167888");
+
+        /*  extracting separately as document_binary_url in populateCaseDetailsWithSingleC21Element()
+            method uses UUID.randomUUID()
+         */
+        assertThat(returnedParameters).extracting("linkToDocStore").isNotEqualTo("");
+    }
+
+    @Test
+    void shouldReturnExactC21NotificationParametersWithMostRecentUploadedDocumentUrl() {
+        CaseDetails caseDetails = populateCaseDetails();
+
+        Map<String, Object> returnedParameters = c21OrderEmailContentProvider.buildC21OrderNotification(caseDetails,
+            LOCAL_AUTHORITY_CODE);
+
+        final String subjectLine = "Jones, " + familyManCaseNumber;
+
+        assertThat(returnedParameters)
+            .extracting("subjectLine", "localAuthorityOrCafcass", "hearingDetailsCallout", "linkToDocStore",
+                "reference", "caseUrl")
+            .containsExactly(subjectLine, "Example Local Authority",
+                subjectLine + ", hearing " + dateFormatterService.formatLocalDateToString(date, FormatStyle.MEDIUM),
+                "http://dm-store:8080/documents/79ec80ec-7be6-493b-b4e6-f002f05b7079/binary", "12345",
+                "/case/" + JURISDICTION + "/" + CASE_TYPE + "/12345");
+    }
+
+    private CaseDetails populateCaseDetailsWithSingleC21Element() {
+        return CaseDetails.builder()
+            .id(167888L)
+            .data(ImmutableMap.of("hearingDetails", createHearingBookings(LocalDate.now()),
+                "c21Orders", ImmutableList.of(
+                    Element.<C21Order>builder()
+                        .value(C21Order.builder()
+                            .orderTitle("Example Order")
+                            .orderDetails(
+                                "Example order details here - Lorem ipsum dolor sit amet, consectetur adipiscing elit")
+                            .judgeAndLegalAdvisor(JudgeAndLegalAdvisor.builder()
+                                .legalAdvisorName("Peter Parker")
+                                .judgeLastName("Judy")
+                                .judgeTitle(HER_HONOUR_JUDGE)
+                                .build())
+                            .document(DocumentReference.builder()
+                                .filename("C21 1.pdf")
+                                .url("http://" + String.join("/", "dm-store:8080", "documents",
+                                    randomUUID().toString()))
+                                .binaryUrl("http://" + String.join("/", "dm-store:8080", "documents",
+                                    randomUUID().toString(), "binary"))
+                                .build())
+                            .build())
+                        .build()),
+                "respondents1", createRespondents(),
+                "familyManCaseNumber", familyManCaseNumber))
+            .build();
+    }
+
+    private CaseDetails populateCaseDetails() {
```

Comment:
Not a great method name. Let's be more descriptive.

Reply:
updated